internal/datanode/compaction/compactor_common.go

```diff
@@ -229,3 +229,45 @@ func mergeFieldBinlogs(base, paths map[typeutil.UniqueID]*datapb.FieldBinlog) {
 		base[fID].Binlogs = append(base[fID].Binlogs, fpath.GetBinlogs()...)
 	}
 }
+
+func bmSerializeWrite(ctx context.Context, io io.BinlogIO, allocator allocator.Interface, writer *SegmentWriter) ([]*datapb.FieldBinlog, error) {
```

Comment:
this method is mostly the same as `bm25SerializerWrite` in `indexnode/task_stats.go`, avoid duplicate code